### PR TITLE
switch from using affine3d to isometry3d

### PR DESCRIPTION
**Pull request by Levi-Armstrong**

### CODE REVIEW

File: tesseract_core/include/tesseract_core/basic_types.h [hunk 1]
```diff
@@ -34,13 +34,15 @@
...
+
 
 namespace tesseract
 {
 typedef Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> TrajArray;
-typedef std::vector<Eigen::Affine3d, Eigen::aligned_allocator<Eigen::Affine3d> > vector_Affine3d;
-typedef std::map<std::string, Eigen::Affine3d> TransformMap;
+typedef std::vector<Eigen::Isometry3d, Eigen::aligned_allocator<Eigen::Isometry3d> > VectorIsometry3d;
+typedef std::vector<Eigen::Vector4d, Eigen::aligned_allocator<Eigen::Vector4d> > VectorVector4d;
+typedef std::map<std::string, Eigen::Isometry3d> TransformMap;
```
Comment (Jmeyer1292) [inline tesseract_core/include/tesseract_core/basic_types.h]: Technically, `std::map` should also take an aligned allocator here. See Eigen [docs](https://eigen.tuxfamily.org/dox/group__TopicStlContainers.html).

Shameless plug, you can use templated aliases to make this code very concise and general. See my [PR](https://github.com/ros/eigen_stl_containers/pull/14). For your convenience:
```c++
template <typename Key, typename Value>
using AlignedMap = std::map<Key, Value, std::less<Key>, Eigen::aligned_allocator<std::pair<const Key, Value>>>; 
```
```c++
template <typename T>
using AlignedVector = std::vector<T, Eigen::aligned_allocator<T>>;
```

**Comment by Levi-Armstrong** [comment]
@Jmeyer1292 Ok to merge

**Comment by Jmeyer1292** [comment]
@Levi-Armstrong I do not have write access to this repo. I can't merge.

**Comment by Levi-Armstrong** [comment]
@Jmeyer1292 Ok, you have write access now on this repository and on trajopt_ros

**Pull request merged**